sway/tree/workspace.c

```diff
@@ -734,7 +734,29 @@ void workspace_add_gaps(struct sway_workspace *ws) {
 		}
 	}
 
-	ws->current_gaps = ws->gaps_outer;
+	// Since we can't rely on outer gaps being small enough for the specific
+	// output we are in, cap them so at least the minimum sane space is left.
+	if (ws->width - ws->gaps_outer.left - ws->gaps_outer.right > MIN_SANE_W) {
+		ws->current_gaps.left = ws->gaps_outer.left;
+		ws->current_gaps.right = ws->gaps_outer.right;
+	} else {
+		int total_gap = MAX(0, ws->width - MIN_SANE_W);
+		double left_gap_frac = ((float) ws->gaps_outer.left /
+			((float) ws->gaps_outer.left + (float) ws->gaps_outer.right));
```

Comment:
This will produce a `SIGFPE` when `ws->gaps_outer.left + ws->gaps_outer.right == 0`

Reply:
Only if you find a workspace that's <= than MIN_SANE_W :) But yeah it should be caught and is simple enough to fix.